src/HTTP.jl

```diff
@@ -297,20 +300,25 @@ HTTP.open("POST", "http://music.com/play") do io
 end
 ```
 """
-request(method::String, url::URI, headers::Headers, body; kw...)::Response =
-    request(HTTP.stack(;kw...), method, url, headers, body; kw...)
 #FIXME consider @nospecialize for `body` ? (other places? in ConnectionPool?)
-
+function request(method::String, url::URI, headers::Headers, body; kw...)::Response
+    request(HTTP.stack(;kw...), method, url, headers, body; kw...)
+end
 
 const nobody = UInt8[]
 
 function request(method, url, h=Header[], b=nobody;
-                 headers=h, body=b, query=nothing, kw...)::Response
+                 headers=h, body=b, query=nothing, custom_stack=nothing, kw...)::Response
```

Comment:
Do we really need this as a kwarg?

Isn't using the `request(custom_stack, method, url, ...)`  method enough?

Reply:
That won't get the pass-through from `get`, `post`, etc. We could add methods for those though?

Reply:
What Eric mentioned is the alternative, I thought this was a more clean solution to give access to the "low" and "high" level calls.

I can go either way on this change.

Reply:
I think adding methods for all of them is better.
Potentialy via an `@eval` loop